pkg/repo/migrations/v3_4.go

```diff
@@ -0,0 +1,28 @@
+package migrations
+
+import (
+	"os"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config"
+	"github.com/bacalhau-project/bacalhau/pkg/repo"
+)
+
+// V3Migration deletes the network store (NATS KV store) so that it may recreated once the node starts.
+// It deletes the store since in v1.3.0 the store contains models.NodeInfo and in v1.3.1 the store
+// contains models.NodeState and these types are incompatible with each other.
+// Full details may be found in https://github.com/bacalhau-project/bacalhau/issues/4024
+var V3Migration = repo.NewMigration(
+	repo.RepoVersion3,
```

Comment:
please add a test scenario for this migration. We already have a test suite for repo migrations and should be trivial to add this one

Reply:
also you might want to validate that migration will be graceful if the store doesn't exist or the path is not configured properly

Reply:
Was waiting for feedback on this vs https://github.com/bacalhau-project/bacalhau/pull/4029 before doing any more work here. But if we decide we like this PR better I will add tests before merging.